crates/core/src/kernel/scalars.rs

```diff
@@ -344,6 +346,14 @@ impl ScalarExt for Scalar {
     }
 }
 
+fn checked(arr: &dyn Array, idx: usize, val: Scalar) -> Scalar {
```

Comment:
@roeap  this function is a good candidate for [inlining](https://nnethercote.github.io/perf-book/inlining.html) for future reference.

There may be a couple others already in the codebase, but good to know about in the future :wink: